hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightSearchEndpoint.java

```diff
@@ -0,0 +1,390 @@
+        if (matchedKeys.size() >= limit) {
+          break;
+        }
+      }
+      return matchedKeys;
+    }
+
+    // Iterate over for bucket and volume level search
+    matchedKeys.putAll(retrieveKeysFromTable(openFileTable, startPrefixObjectPath, limit, prevKey));
+    return matchedKeys;
+  }
+
+  /**
+   * Finds all subdirectories under a parent directory in an FSO bucket. It builds
+   * a list of paths for these subdirectories. These sub-directories are then used
+   * to search for open files in the openFileTable.
+   *
+   * How it works:
+   * - Starts from a parent directory identified by parentId.
+   * - Looks through all child directories of this parent.
+   * - For each child, it creates a path that starts with volumeID/bucketID/parentId,
+   * following our openFileTable format
+   * - Adds these paths to a list and explores each child further for more subdirectories.
+   *
+   * @param parentId The ID of the directory we start exploring from.
+   * @param subPaths A list where we collect paths to all subdirectories.
+   * @param names    An array with at least two elements: the first is volumeID and
+   *                 the second is bucketID. These are used to start each path.
+   * @throws IOException If there are problems accessing directory information.
+   */
+  private void gatherSubPaths(long parentId, List<String> subPaths,
+                              String[] names) throws IOException {
+    // Fetch the NSSummary object for parentId
+    NSSummary parentSummary = reconNamespaceSummaryManager.getNSSummary(parentId);
+    if (parentSummary == null) {
+      return;
+    }
+    long volumeID = Long.parseLong(names[0]);
+    long bucketID = Long.parseLong(names[1]);
+    Set<Long> childDirIds = parentSummary.getChildDir();
+    for (Long childId : childDirIds) {
+      // Fetch the NSSummary for each child directory
+      NSSummary childSummary = reconNamespaceSummaryManager.getNSSummary(childId);
+      if (childSummary != null) {
+        String subPath = constructObjectPathWithPrefix(volumeID, bucketID, childId);
+        // Add to subPaths
+        subPaths.add(subPath);
+        // Recurse into this child directory
+        gatherSubPaths(childId, subPaths, names);
+      }
+    }
+  }
+
+
+  /**
+   * Converts a key prefix into an object path for FSO buckets, using IDs.
+   * <p>
+   * This method transforms a user-provided path (e.g., "volume/bucket/dir1") into
+   * a database-friendly format ("/volumeID/bucketID/ParentId/") by replacing names
+   * with their corresponding IDs. It simplifies database queries for FSO bucket operations.
+   *
+   * @param prevKeyPrefix The path to be converted, not including key or directory names/IDs.
+   * @return The object path as "/volumeID/bucketID/ParentId/" or an empty string if an error occurs.
+   * @throws IOException If database access fails.
+   */
+  public String convertToObjectPath(String prevKeyPrefix) throws IOException, IllegalArgumentException {
```

Comment:
In what use case, it will throw `IllegalArgumentException`, pls check, I have a doubt, and even if this method throws in some scenario, then pls write javadoc comment.